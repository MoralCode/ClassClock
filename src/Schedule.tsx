import React, { Component } from 'react';

class Schedule extends Component {
   
    // document.getElementById("schoolName").innerHTML = schools[selectedSchoolIndex].fullName;
    // document.getElementById("scheduleDisplay").innerHTML = getCurrentScheduleName() + " schedule.";

    // populateScheduleTable();


    //input:  onchange="updateSettings()"
    render() {
        return (
            <div>
                <a className="navbutton" href="index.html"><i className="fas fa-home" /></a>
                <br />
                <h1 className="centered topSpace bottomSpace" id="schoolName" />
                <p className="centered bottomSpace" id="scheduleDisplay" />
                <table id="scheduleTable" className="centeredInline topSpace" />
            </div>
        );
    }
}