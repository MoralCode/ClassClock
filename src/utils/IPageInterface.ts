export default interface IPageInterface {
    dispatch: any;
}
