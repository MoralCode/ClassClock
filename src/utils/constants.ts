
export const pages = {
    main: "/",
    fullSchedule: "/schedule",
    settings: "/settings",
    fullScheduleLegacy: "/schedule.html",
    settingsLegacy: "/settings.html",
    selectSchool: "/select",
    loginCallback: "/admin/callback",
    welcome: "/welcome",
    admin: "/admin",
    adminBellSchedule: "/admin/bellschedule"
};

export const URLs = {
    github: "https://github.com/MoralCode/ClassClock",
    slack:
        "https://join.slack.com/t/classclock/shared_invite/enQtNTE0MDkyNzAwNzU3LWNhMGUwODU2ZjhkYTYxMTgzNDE1OWEyMGY2OGNiNTBhOWM5NDVhZGUzNDVlNzRiZTE3NTNmODFjYWNkNDhmMDU",
    twitter: "https://twitter.com/classclockapp",
    instagram: "https://www.instagram.com/classclockapp/",
    bluesky: "https://bsky.app/profile/classclockapp.bsky.social",
    discord: "https://discord.classclock.app"
};
