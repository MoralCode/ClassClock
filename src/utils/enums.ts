export enum TimeStates {
    DAY_OFF = "day off",
    OUTSIDE_SCHOOL_HOURS = "outside school hours",
    SCHOOL_IN_CLASS_OUT = "school is in session, but class is not",
    CLASS_IN_SESSION = "class is in session"
}

export enum TimeComparisons {
    IS_BEFORE = -1, //"before",
    IS_DURING_OR_EXACTLY = 0, //"during/exactly",
    IS_AFTER = 1 //"after"
}
