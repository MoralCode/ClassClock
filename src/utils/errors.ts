// errors.ts
// This file contains classes that represent error conditions
// @author Adrian Edwards



/**
 * This custom Error class is intened to be used for passing a minimum wait 
 * time to the .catch() handlers responsible for retrying requests so requests 
 * arent made during a rate-limit coolown period
 * 
 * This class was created with the help of https://javascript.info/custom-errors
 *
 * @export
 * @class RateLimitError
 * @extends {Error}
 */
export class RateLimitError extends Error {
	wait: number;
	constructor(message:string, msToWait: number) {
		super(message);
		this.name = "RateLimitError";
		this.wait = msToWait;
	}
}