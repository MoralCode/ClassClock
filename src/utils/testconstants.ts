// This is not actually a unit test, just a central place to hold frequently-used constants that are often reused across tests

import Time from "../@types/time";
import School from "../@types/school";
import BellSchedule from "../@types/bellschedule";
import ClassPeriod from "../@types/classperiod";





export const currentDate = new Date("2019-07-28T07:37:50.634Z");

export const className = "First Period";

export const beforeClass = new Time(8, 0);
export const beforeClassString = "8:00"
export const startTime = new Time(8, 25);
export const startTimeString = "8:25";
export const classDuration = new Time(1, 30);
export const duringClass = new Time(9, 0);
export const duringClassString = "9:00";
export const endTime = new Time(9, 55);
export const endTimeString = "9:55";
export const afterClass = new Time(10, 0);
export const afterClassString = "10:00";

export const classPeriod = new ClassPeriod(className, startTime, endTime, currentDate);

export const classPeriodJSON = {
           name: className,
           startTime: startTimeString,
           endTime: endTimeString,
           creationDate: currentDate
       };

export const bellScheduleId = "1";
export const bellScheduleName = "Regular Schedule";
export const bellScheduleEndpoint = "/path/to/sched";
export const bellScheduleDisplayName = "Display Name";


export const bellSchedule = new BellSchedule(
    bellScheduleId,
    bellScheduleName,
    bellScheduleEndpoint,
    [
        new Date("2019-07-28T07:37:50.634Z"),
        new Date("2019-07-29T07:38:10.979Z"),
        new Date("2019-07-23T07:38:28.263Z")
    ],
    [classPeriod],
    currentDate
);

export const bellScheduleJSON = {
           id: bellScheduleId,
           name: bellScheduleName,
           endpoint: bellScheduleEndpoint,
           dates: [
               "2019-07-28T07:37:50.634Z",
               "2019-07-29T07:38:10.979Z",
               "2019-07-23T07:38:28.263Z"
           ],
           classes: [classPeriodJSON],
           lastModified: currentDate
       };


export const schoolId = "10";
export const schoolName = "The High School";
export const schoolAcronym = "THS";
export const schoolEndpoint = "/some/endpoint";
export const schoolTimezone = "America/Cupertino";
export const passingPeriodName = "Transition Time";

export const school = new School(
    schoolId,
    schoolName,
    schoolAcronym,
    schoolEndpoint,
    schoolTimezone,
    [bellSchedule],
    passingPeriodName,
    currentDate,
    currentDate
);

export const schoolJSON = {
           id: schoolId,
           name: schoolName,
           acronym: schoolAcronym,
           endpoint: schoolEndpoint,
           timezone: schoolTimezone,
           schedules: [bellScheduleJSON],
           passingPeriodName,
           creationDate: currentDate,
           lastModified: currentDate
       };
