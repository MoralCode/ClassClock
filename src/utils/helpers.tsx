import Time from "../@types/time";
import School from "../@types/school";
import { TimeComparisons, TimeStates } from "./enums";
import ClassPeriod from "../@types/classperiod";

export function getValueIfKeyInList(list: string[], object: any) {
    for (const key of list) {
        if (object.hasOwnProperty(key)) {
            return object[key];
        }
    }
}

export function deconstructJsonApiResource(json: any) {
    const data = {
        type: json.type,
        id: json.id,
        ...(json.links !== undefined && {
            endpoint: json.links.self
        })
    };
    return Object.assign({}, data, json.attributes);
}

export function getCurrentDate() {
    return new Date();
}

/**
 * @returns a flag that represents the current chunk of time categorically
 */
export function getTimeStateForDateAtSchool(date: Date, school: School) {
    const currentBellSchedule = school.getScheduleForDate(date);

    //there is no schedule that applies today
    if (!currentBellSchedule) {
        return TimeStates.DAY_OFF;
    }

    const currentClassPeriod = currentBellSchedule.getClassPeriodForTime(
        Time.fromDate(date)
    );

    //it is a school day but it is not school hours
    if (!school.isInSession(date)) {
        return TimeStates.OUTSIDE_SCHOOL_HOURS;
    }

    //the current time lies between the start of the first schedules class and the end of the last
    else if (school.isInSession(date) && !currentClassPeriod) {
        return TimeStates.SCHOOL_IN_CLASS_OUT;
    }

    //the current time lies within a scheduled class period
    else if (currentClassPeriod !== undefined) {
        return TimeStates.CLASS_IN_SESSION;
    }
}

/**
 * @returns the next relevent time to count down to
 */
export function getNextImportantTime(
    date: Date,
    school: School
): [ClassPeriod, Time] | undefined {
    const currentBellSchedule = school.getScheduleForDate(date);

    //there is no schedule that applies today
    if (!currentBellSchedule) {
        return;
    }

    currentBellSchedule.getAllClasses().forEach((value: ClassPeriod) => {
        for (const time of [value.getStartTime(), value.getEndTime()]) {
            if (time.getMillisecondsTo(Time.fromDate(date)) >= 0) {
                return [value, time];
            }
        }
    });
}

/**
 * This export function checks if the current time is between the two given times
 * This is useful for checking which class period you are currently in or for checking if school is in session.
 *
 * @param {*} checkTime the time that the check results are returned for
 * @param {*} startTime the start time of the range to check
 * @param {*} endTimethe the end time of the range to check
 *
 * @returns -1 if checkTime is before range, 0 if checkTime is within range, 1 if checkTime is after range
 */
export function checkTimeRange(checkTime: Time, startTime: Time, endTime: Time) {
    if (startTime.getMillisecondsTo(endTime) <= 0) {
        //theres a problem
    }
    const startCheck = checkTime.getMillisecondsTo(startTime);
    const endCheck = checkTime.getMillisecondsTo(endTime);

    if (startCheck > 0) {
        return TimeComparisons.IS_BEFORE;
    } else if (endCheck < 0) {
        return TimeComparisons.IS_AFTER;
    } else {
        return TimeComparisons.IS_DURING_OR_EXACTLY;
    }
}
