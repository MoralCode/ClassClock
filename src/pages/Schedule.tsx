import React, { Component } from "react";
import { connect } from "react-redux";
import { push } from "redux-first-routing";
import IPageInterface from "../utils/IPageInterface";

import "../global.css";
import School from "../@types/school";
import { IState } from "../store/schools/types";
import Link from "../components/Link";
import { pages } from "../utils/constants";
import Icon from "../components/Icon";
import { getCurrentDate, sortClassesByStartTime } from "../utils/helpers";
import ClassPeriod from "../@types/classperiod";
import ScheduleEntry from "../components/ScheduleEntry/ScheduleEntry";
import List from "../components/List/List";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faHome } from "@fortawesome/free-solid-svg-icons";

export interface IAppProps {
    selectedSchool: {
        isFetching: boolean;
        didInvalidate: false;
        data: School;
    };
    dispatch: any;
}

const Schedule = (props: IAppProps) => {
    let content: JSX.Element = <></>;
    const currentSchedule = props.selectedSchool.data.getScheduleForDate(
        getCurrentDate()
    );

    switch (currentSchedule) {
        case undefined:
            props.dispatch(push(pages.selectSchool));
            break;
        case null:
            content = <p>No School Today</p>;
            break;
        default:
            content = (
                <>
                    <p>{currentSchedule.getName()}</p>
                    {/* <List items={scheduleItems} /> */}
                    <table>
                        <thead>
                            <tr>
                                <td>
                                    <b>Class</b>
                                </td>
                                <td>
                                    <b>Time</b>
                                </td>
                            </tr>
                        </thead>
                        <tbody>
                            {sortClassesByStartTime(currentSchedule.getAllClasses()).map(
                                (value: ClassPeriod) => (
                                    <>
                                        <tr>
                                            <td>{value.getName()}</td>
                                            <td>
                                                {value.getStartTime().toString()} -{" "}
                                                {value.getEndTime().toString()}
                                            </td>
                                        </tr>
                                    </>
                                )
                            )}
                        </tbody>
                    </table>
                </>
            );
            break;
    }

    return (
        <div>
            <Link
                className="cornerNavButton cornerNavLeft smallIcon"
                // tslint:disable-next-line: jsx-no-lambda
                destination={() => props.dispatch(push(pages.main))}
            >
                <FontAwesomeIcon icon={faHome} />
            </Link>
            <br />
            <p style={{ fontSize: "30px" }}>
                <b>{props.selectedSchool.data.getName()}</b>
            </p>

            {content}
        </div>
    );
};

const mapStateToProps = (state: IState) => {
    const { selectedSchool } = state;
    selectedSchool.data = School.fromJson(selectedSchool.data);
    return { selectedSchool };
};

export default connect(mapStateToProps)(Schedule);
