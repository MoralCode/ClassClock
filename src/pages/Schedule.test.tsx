import React from 'react';
import ReactDOM from 'react-dom';
import renderer from "react-test-renderer";
import { Schedule } from './Schedule';
import { school, beforeSchoolHours, noSchool, inClass, betweenClass, afterSchoolHours } from '../utils/testconstants';
import MockDate from 'mockdate';

// make a fake dispatch function, it doesnt need to do anything, because we assume that has already been tested in the redux library
const mockDispatch = jest.fn();

const mockState = {
    isFetching: false,
    didInvalidate: false,
    data: school
};

const rawComponent = 
    <Schedule selectedSchool={mockState} dispatch={mockDispatch} />



describe("Schedule", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(rawComponent, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("shows the day's schedule regardless of the time", () => {
      MockDate.set(beforeSchoolHours.toJSDate());
      let tree = renderer.create(rawComponent).toJSON();
      expect(tree).toMatchSnapshot();
 
      MockDate.set(noSchool.toJSDate());
      tree = renderer.create(rawComponent).toJSON();
      expect(tree).toMatchSnapshot();
  
      MockDate.set(inClass.toJSDate());
      tree = renderer.create(rawComponent).toJSON();
      expect(tree).toMatchSnapshot();
  
      MockDate.set(betweenClass.toJSDate());
      tree = renderer.create(rawComponent).toJSON();
      expect(tree).toMatchSnapshot();
 
      MockDate.set(afterSchoolHours.toJSDate());
      tree = renderer.create(rawComponent).toJSON();
      expect(tree).toMatchSnapshot();
  });

});


