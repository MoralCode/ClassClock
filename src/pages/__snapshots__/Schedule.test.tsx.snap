// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule shows the day's schedule regardless of the time 1`] = `
<div>
  <a
    className="cornerNavButton cornerNavTop cornerNavLeft smallIcon"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-house "
      data-icon="house"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M575.8 255.5C575.8 273.5 560.8 287.6 543.8 287.6H511.8L512.5 447.7C512.5 450.5 512.3 453.1 512 455.8V472C512 494.1 494.1 512 472 512H456C454.9 512 453.8 511.1 452.7 511.9C451.3 511.1 449.9 512 448.5 512H392C369.9 512 352 494.1 352 472V384C352 366.3 337.7 352 320 352H256C238.3 352 224 366.3 224 384V472C224 494.1 206.1 512 184 512H128.1C126.6 512 125.1 511.9 123.6 511.8C122.4 511.9 121.2 512 120 512H104C81.91 512 64 494.1 64 472V360C64 359.1 64.03 358.1 64.09 357.2V287.6H32.05C14.02 287.6 0 273.5 0 255.5C0 246.5 3.004 238.5 10.01 231.5L266.4 8.016C273.4 1.002 281.4 0 288.4 0C295.4 0 303.4 2.004 309.5 7.014L564.8 231.5C572.8 238.5 576.9 246.5 575.8 255.5L575.8 255.5z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <br />
  <p
    style={
      Object {
        "fontSize": "30px",
      }
    }
  >
    <b>
      The High School
    </b>
  </p>
  <p>
    Regular Schedule
  </p>
  <table>
    <thead>
      <tr>
        <td>
          <b>
            Class
          </b>
        </td>
        <td>
          <b>
            Time
          </b>
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          First Period
        </td>
        <td>
          08:25:00
           -
           
          09:55:00
        </td>
      </tr>
      <tr>
        <td>
          Second Period
        </td>
        <td>
          10:05:00
           -
           
          11:35:00
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Schedule shows the day's schedule regardless of the time 2`] = `
<div>
  <a
    className="cornerNavButton cornerNavTop cornerNavLeft smallIcon"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-house "
      data-icon="house"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M575.8 255.5C575.8 273.5 560.8 287.6 543.8 287.6H511.8L512.5 447.7C512.5 450.5 512.3 453.1 512 455.8V472C512 494.1 494.1 512 472 512H456C454.9 512 453.8 511.1 452.7 511.9C451.3 511.1 449.9 512 448.5 512H392C369.9 512 352 494.1 352 472V384C352 366.3 337.7 352 320 352H256C238.3 352 224 366.3 224 384V472C224 494.1 206.1 512 184 512H128.1C126.6 512 125.1 511.9 123.6 511.8C122.4 511.9 121.2 512 120 512H104C81.91 512 64 494.1 64 472V360C64 359.1 64.03 358.1 64.09 357.2V287.6H32.05C14.02 287.6 0 273.5 0 255.5C0 246.5 3.004 238.5 10.01 231.5L266.4 8.016C273.4 1.002 281.4 0 288.4 0C295.4 0 303.4 2.004 309.5 7.014L564.8 231.5C572.8 238.5 576.9 246.5 575.8 255.5L575.8 255.5z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <br />
  <p
    style={
      Object {
        "fontSize": "30px",
      }
    }
  >
    <b>
      The High School
    </b>
  </p>
  <p>
    No School Today
  </p>
</div>
`;

exports[`Schedule shows the day's schedule regardless of the time 3`] = `
<div>
  <a
    className="cornerNavButton cornerNavTop cornerNavLeft smallIcon"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-house "
      data-icon="house"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M575.8 255.5C575.8 273.5 560.8 287.6 543.8 287.6H511.8L512.5 447.7C512.5 450.5 512.3 453.1 512 455.8V472C512 494.1 494.1 512 472 512H456C454.9 512 453.8 511.1 452.7 511.9C451.3 511.1 449.9 512 448.5 512H392C369.9 512 352 494.1 352 472V384C352 366.3 337.7 352 320 352H256C238.3 352 224 366.3 224 384V472C224 494.1 206.1 512 184 512H128.1C126.6 512 125.1 511.9 123.6 511.8C122.4 511.9 121.2 512 120 512H104C81.91 512 64 494.1 64 472V360C64 359.1 64.03 358.1 64.09 357.2V287.6H32.05C14.02 287.6 0 273.5 0 255.5C0 246.5 3.004 238.5 10.01 231.5L266.4 8.016C273.4 1.002 281.4 0 288.4 0C295.4 0 303.4 2.004 309.5 7.014L564.8 231.5C572.8 238.5 576.9 246.5 575.8 255.5L575.8 255.5z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <br />
  <p
    style={
      Object {
        "fontSize": "30px",
      }
    }
  >
    <b>
      The High School
    </b>
  </p>
  <p>
    Regular Schedule
  </p>
  <table>
    <thead>
      <tr>
        <td>
          <b>
            Class
          </b>
        </td>
        <td>
          <b>
            Time
          </b>
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          First Period
        </td>
        <td>
          08:25:00
           -
           
          09:55:00
        </td>
      </tr>
      <tr>
        <td>
          Second Period
        </td>
        <td>
          10:05:00
           -
           
          11:35:00
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Schedule shows the day's schedule regardless of the time 4`] = `
<div>
  <a
    className="cornerNavButton cornerNavTop cornerNavLeft smallIcon"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-house "
      data-icon="house"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M575.8 255.5C575.8 273.5 560.8 287.6 543.8 287.6H511.8L512.5 447.7C512.5 450.5 512.3 453.1 512 455.8V472C512 494.1 494.1 512 472 512H456C454.9 512 453.8 511.1 452.7 511.9C451.3 511.1 449.9 512 448.5 512H392C369.9 512 352 494.1 352 472V384C352 366.3 337.7 352 320 352H256C238.3 352 224 366.3 224 384V472C224 494.1 206.1 512 184 512H128.1C126.6 512 125.1 511.9 123.6 511.8C122.4 511.9 121.2 512 120 512H104C81.91 512 64 494.1 64 472V360C64 359.1 64.03 358.1 64.09 357.2V287.6H32.05C14.02 287.6 0 273.5 0 255.5C0 246.5 3.004 238.5 10.01 231.5L266.4 8.016C273.4 1.002 281.4 0 288.4 0C295.4 0 303.4 2.004 309.5 7.014L564.8 231.5C572.8 238.5 576.9 246.5 575.8 255.5L575.8 255.5z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <br />
  <p
    style={
      Object {
        "fontSize": "30px",
      }
    }
  >
    <b>
      The High School
    </b>
  </p>
  <p>
    Regular Schedule
  </p>
  <table>
    <thead>
      <tr>
        <td>
          <b>
            Class
          </b>
        </td>
        <td>
          <b>
            Time
          </b>
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          First Period
        </td>
        <td>
          08:25:00
           -
           
          09:55:00
        </td>
      </tr>
      <tr>
        <td>
          Second Period
        </td>
        <td>
          10:05:00
           -
           
          11:35:00
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Schedule shows the day's schedule regardless of the time 5`] = `
<div>
  <a
    className="cornerNavButton cornerNavTop cornerNavLeft smallIcon"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-house "
      data-icon="house"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M575.8 255.5C575.8 273.5 560.8 287.6 543.8 287.6H511.8L512.5 447.7C512.5 450.5 512.3 453.1 512 455.8V472C512 494.1 494.1 512 472 512H456C454.9 512 453.8 511.1 452.7 511.9C451.3 511.1 449.9 512 448.5 512H392C369.9 512 352 494.1 352 472V384C352 366.3 337.7 352 320 352H256C238.3 352 224 366.3 224 384V472C224 494.1 206.1 512 184 512H128.1C126.6 512 125.1 511.9 123.6 511.8C122.4 511.9 121.2 512 120 512H104C81.91 512 64 494.1 64 472V360C64 359.1 64.03 358.1 64.09 357.2V287.6H32.05C14.02 287.6 0 273.5 0 255.5C0 246.5 3.004 238.5 10.01 231.5L266.4 8.016C273.4 1.002 281.4 0 288.4 0C295.4 0 303.4 2.004 309.5 7.014L564.8 231.5C572.8 238.5 576.9 246.5 575.8 255.5L575.8 255.5z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <br />
  <p
    style={
      Object {
        "fontSize": "30px",
      }
    }
  >
    <b>
      The High School
    </b>
  </p>
  <p>
    Regular Schedule
  </p>
  <table>
    <thead>
      <tr>
        <td>
          <b>
            Class
          </b>
        </td>
        <td>
          <b>
            Time
          </b>
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          First Period
        </td>
        <td>
          08:25:00
           -
           
          09:55:00
        </td>
      </tr>
      <tr>
        <td>
          Second Period
        </td>
        <td>
          10:05:00
           -
           
          11:35:00
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
